generate/generate.go

```diff
@@ -543,6 +553,16 @@ func checkGeneratedFilesConflicts(genfiles []genfile) error {
 	return nil
 }
 
+func checkGeneratedFilesPaths(genfiles []genfile) error {
+	for _, gen := range genfiles {
+		if strings.Contains(gen.name, "/") {
```

Comment:
I think we should check for `filepath.Separator` otherwise on windows `dir\test.tf` will be valid, or we convert everything to ForwardSlash before calling the generation.

Reply:
Will do + add tests for this

Reply:
hmm actually filepath.ToSlash introduces the same problem, when using `\` on linux it will not replace it with slash:

https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/path/filepath/path.go;l=167

Reply:
OK gave it a shot by checking for both separators with Contains. Funny enough, using filepath.Separator would not work at all since on Windows `/` is also a separator:

https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/os/path_windows.go;l=13;bpv=1;bpt=0

So we would allow `/` on windows but that would be interpreted as a valid separator/dir for Windows (apparently, Im quite outdated on my windows skills :laughing: )